src/components/BookContainer.js

```diff
@@ -0,0 +1,33 @@
+import React from 'react';
+import { useSelector, useDispatch } from 'react-redux';
+import { addBook } from '../redux/books/booksSlice';
+import Book from './Book';
+import Button from './Button';
+
+const BookContainer = () => {
+  const books = useSelector((state) => state.books.books);
+  const dispatch = useDispatch();
+
+  const handleAddBook = () => {
+    const newBook = {
+      item_id: `item${books.length + 1}`,
+      title: 'New Book Title',
+      author: 'New Author Name',
+      category: 'Fiction',
+    };
+    dispatch(addBook(newBook));
+  };
+
+  return (
+    <div>
+      {books.map((book) => (
+        <div key={book.item_id}>
+          <Book book={book} />
+        </div>
+      ))}
+      <Button buttonClass="add-btn" type="button" onClick={handleAddBook} text="Add Book" />
+    </div>
+  );
```

Comment:
- [ ] On the page I am not able to see a form where I can add a new book that should be updated through the onChange event when I click on the button I am getting please fix that
<img width="1426" alt="Screenshot 2023-08-03 at 17 18 00" src="https://github.com/akbarsazish/Bookstore-React/assets/44745855/e717fde5-45f1-4314-ac63-6f37fe6b5407">